### DISCUSSION TITLE
Export Numerous Tours in GPX

### DISCUSSION THREAD

**Discussion by mff73** [mytourbook/mytourbook#1402] [Q&A] [answered]
Hello,
I need to export numerous tours (one year) into a unique GPX file.
When selecting a certain amount of tours (limit is not clear) I can right click and get the export function available
![image](https://github.com/user-attachments/assets/fc641f18-bb40-4cb0-b8bf-a48383cbc4fc)
but when selecting lots more tours (example my full 2023 year (=206 tours)) nothing open on right click and a furtive popup is appearing and disapearing. But no context menu to export.

https://github.com/user-attachments/assets/8ddd5e9d-015f-41a9-b99d-719d86743161

except exporting fewer tours at once, is there something known/limited ? any workaround ?
Thanks.

Version 24.8.0

**Comment by stvzldn** [comment]
It would appear that it is processing the request based "traitement en cours" on the transient popup which shows 206 "Chargement des parcours"
Click on ~8 seconds into the video to view the popup

**Comment by wolfgang-ch** [comment]
For me it works


https://github.com/user-attachments/assets/0eb334c7-e639-4e82-818d-3a54fc085880

**Comment by mff73** [comment]
Thanks.
Is there any detailed log I should enable to go further?
There is nothing appearing in mine after my non successful right clicks. 🤔

**Reply by wolfgang-ch** [reply]
To see any logged infos, you have to run MT with a console

`"C:\Program Files\MyTourbook\mytourbook.exe" -debug -consolelog`

This is my production MT commandline, I always run MT with a console to see any possible bugs as early as possible.

`"C:\Program Files\MyTourbook\mytourbook.exe" -nl en_GB -debug -consolelog -showlocation -vmargs -xxxXmx4g -xxxDcreateRandomTourId -xxxDlogFitData -Dswt.autoScale=200`

xxx disables a command line argument

**Reply by mff73** [reply]
thanks.
Here is my console file. No error appearing after startup.
I tried as many things as possible.
When i have the time to see this popup window, the content is : 
![image](https://github.com/user-attachments/assets/05d04b91-b739-4136-8b14-46f61631fa0a)

Another thing which i don't explain : selecting all tours with my current filter (all types between two dates) with CTRL+A, and my list contains 209 tours in this example, but below on the map, only 195 are displayed. Is it linked to the fact that some of my tours do not have GPS data (thus not displayed on the map) ?

![image](https://github.com/user-attachments/assets/592f361c-6690-4060-a2f5-cf1deee6454a)

[console log 24.8.0 v1.txt](https://github.com/user-attachments/files/16861748/console.log.24.8.0.v1.txt)

**Reply by FJBDev** [reply]
> thanks. Here is my console file. No error appearing after startup. I tried as many things as possible. When i have the time to see this popup window, the content is : 


It seems that in the same view you have the 2D map open. Can you try again but with ONLY the Tour Book view open ?
My hunch is that selecting all those tours trigger the map painting that takes a long time.

**Comment by wolfgang-ch** [comment]
The log is OK, all errors above line 7267 can be ignored, I've no idea how to solve them but they do not hurt.

![image](https://github.com/user-attachments/assets/40b49f65-efaa-4247-a986-c417f28718da)


> Another thing which i don't explain : selecting all tours with my current filter (all types between two dates) with CTRL+A, and my list contains 209 tours in this example, but below on the map, only 195 are displayed. Is it linked to the fact that some of my tours do not have GPS data (thus not displayed on the map) ?

You have seen is correctly, the map ignores all tours which has no geo location.


> I tried as many things as possible.

Have you tried to export only one and/or only a few tours ?

**Reply by mff73** [reply]
I can export 160 or sometimes more tours correctly. 
There seems to be a limit for my wife (160) or my tour list (170)
I just manage to get option for 430 tours, but can't replicate it 😂

**Comment by mff73** [comment, marked as answer]
thanks to everyone.
It seems that my issue is solved with changing to the "embedded server" instead of "network server"

![image](https://github.com/user-attachments/assets/c12b6b1a-6f50-492e-ba87-8c1f6ad04d2e)

even with map enabled, i can export """as much"""" tours i need.

**Reply by wolfgang-ch** [reply]
I could not reproduce this issue but I discovered `null` exceptions which are now fixed in https://github.com/mytourbook/mytourbook/commit/4d92aeb0aa290f60995327a59e13fdf1cc6264a1

**Reply by mff73** [reply]
Cool. If I remember, I will test this when released 😊.
